packages/server/src/effects/logging.ts

```diff
@@ -0,0 +1,23 @@
+import type { SimulationState } from 'src/interfaces';
+import { Effect } from '../effect';
+import { assert } from 'assert-ts';
+import { spawn, Task } from 'effection';
+
+export const loggingEffect = (): Effect<SimulationState> => slice => function*(scope) {
+  let task: Task = yield spawn();
+
+  yield slice.slice('debug').forEach(function*(shouldLogErrors) {
+    if (shouldLogErrors) {
+      yield task.halt();
+      task = yield spawn(function* () {
+        yield slice.filter(({ status }) => status === 'failed').forEach(function *(state) {
+          assert(state.status === 'failed');
+
+          console.error(state.error);
+        });
+      }).within(scope);
+    } else if (!shouldLogErrors) {
+      task.halt();
```

Comment:
Since we halt the task in both cases, it looks like we can just do that at the top once, and only have a single branch if for the positive case.

Reply:
👍